apps/events-helsinki/i18nRoutes.config.js

```diff
@@ -20,13 +20,28 @@ const i18nRoutes = {
     { source: '/sidor/:slug*', locale: 'sv' },
   ],
   '/accessibility': [
-    { source: '/saavutetavuusseloste', locale: 'fi' },
-    { source: '/tillgäntglighetsutlåtande', locale: 'sv' },
+    { source: '/saavutettavuusseloste', locale: 'fi' },
+    { source: '/tillgänglighetsutlåtande', locale: 'sv' },
   ],
   '/about': [
     { source: '/tietoa', locale: 'fi' },
     { source: '/kunskap', locale: 'sv' },
   ],
 };
 
+const i18nRoutesPreviousVersions = {
+  '/fi': [{ source: '/home', locale: 'fi' }],
+  '/sv': [{ source: '/home', locale: 'sv' }],
+  '/en': [{ source: '/home', locale: 'en' }],
+  '/fi/haku': [{ source: '/events', locale: 'fi' }],
+  '/sv/sok': [{ source: '/events', locale: 'sv' }],
+  '/en/search': [{ source: '/events', locale: 'en' }],
+};
+
+const i18nRoutes = Object.assign(
+  {},
+  i18nRoutesPreviousVersions,
+  i18nRoutesCurrentVersion
+);
```

Comment:
`i18nRoutes` is used by the `next.config.js` and it is mapped to `rewrites`:

```
const i18nRoutes = require('./i18nRoutes.config');

/**
 * @type {import('next').NextConfig}
 */
const nextConfig = {
  reactStrictMode: true,
  sassOptions: {
    includePaths: componentsStylePath,
  },
  productionBrowserSourceMaps: !disableSourceMaps,
  i18n,
  async rewrites() {
    return Object.entries(i18nRoutes).flatMap(([destination, sources]) =>
      sources.map(({ source, locale }) => ({
        destination,
        source: `/${locale}${source}`,
        locale: false,
      }))
    );
  },
```

Instead of using rewrites, we should be using redirects (https://nextjs.org/docs/api-reference/next.config.js/redirects)! It's also mentioned in the Jira-ticket.

Reply:
Okay, I found your discussion from the Jira ticket that the redirects didn't work properly. I would like to self try these also, because it should be quite common and basic configuration, that is used by a lot of developers.

Reply:
I tried adding [redirects](https://nextjs.org/docs/api-reference/next.config.js/redirects) into the next.config.js but it didn't work. Tried removing the middleware.js, tried removing the rewrites and leaving the redirects, neither helped as far as I remember. No idea what's causing it. Good luck!

Reply:
It sounds like you have tried just the right things 👍 . The NextJS redirect should still do something, so I will study what it really should do and when it should do that 😄 I'll sure report the results ASAP

Reply:
I got the redirects working in https://github.com/City-of-Helsinki/events-helsinki-monorepo/pull/250

Reply:
I compared how you defined the redirects in PR #250 to the way I tried them out and it is the `locale` parameter which is different. I didn't use the locale parameter at all, tried it out now and yeah, it seems that locale=undefined means locale will be added to the redirects automatically, and the [documentation ](https://nextjs.org/docs/api-reference/next.config.js/redirects#redirects-with-i18n-support) also supports that:
> When leveraging [i18n support](https://nextjs.org/docs/advanced-features/i18n-routing) with redirects each source and destination is automatically prefixed to handle the configured locales unless you add locale: false to the redirect. If locale: false is used you must prefix the source and destination with a locale for it to be matched correctly.

e.g.
```typescript
  async redirects() {
    return [
      { source: '/fi/home', destination: '/fi', permanent: false },
      { source: '/sv/home',
        destination: '/sv',
        locale: undefined,
        permanent: false,
      },
      {
        source: '/en/home',
        destination: '/en',
        locale: false,
        permanent: false,
      },
    ];
  },
```
gives with "yarn build --debug":
```
Redirects

┌ source: /:path+/
├ destination: /:path+
└ permanent: true

┌ source: /default/fi/home
├ destination: /fi
└ permanent: false

┌ source: /:nextInternalLocale(default|fi|sv|en)/fi/home
├ destination: /:nextInternalLocale/fi
└ permanent: false

┌ source: /default/sv/home
├ destination: /sv
└ permanent: false

┌ source: /:nextInternalLocale(default|fi|sv|en)/sv/home
├ destination: /:nextInternalLocale/sv
└ permanent: false

┌ source: /en/home
├ destination: /en
└ permanent: false
```
So of the above only the redirect with `locale: false` works.

So ultimately, using `locale: false` was needed to make the redirects work.